hstream/src/HStream/Server/Handler.hs

```diff
@@ -265,20 +266,24 @@ handleShowPlan ServerContext{..} showObject =
       return (ServerNormalResponse resp [] StatusOk "")
     _ -> returnErrRes "not Supported"
 
-handleTerminatePlan :: ServerContext -> TerminationSelection
-  -> IO (ServerResponse 'Normal CommandQueryResponse)
-handleTerminatePlan ServerContext{..} (OneQuery qid) = do
+handleTerminate :: ServerContext -> TerminationSelection
+  -> IO ()
+handleTerminate ServerContext{..} (OneQuery qid) = do
   hmapQ <- readMVar runningQueries
   case HM.lookup qid hmapQ of Just tid -> killThread tid; _ -> throwIO QueryTerminatedOrNotExist
   P.withMaybeZHandle zkHandle $ P.setQueryStatus qid P.Terminated
-  swapMVar runningQueries (HM.delete qid hmapQ)
-  return (ServerNormalResponse genSuccessQueryResponse [] StatusOk  "")
-handleTerminatePlan ServerContext{..} AllQuery = do
+  void $ swapMVar runningQueries (HM.delete qid hmapQ)
+handleTerminate ServerContext{..} AllQuery = do
   hmapQ <- readMVar runningQueries
   mapM_ killThread $ HM.elems hmapQ
   let f qid = P.withMaybeZHandle zkHandle $ P.setQueryStatus qid P.Terminated
   mapM_ f $ HM.keys hmapQ
-  swapMVar runningQueries HM.empty
+  void $ swapMVar runningQueries HM.empty
+
+handleTerminatePlan :: ServerContext -> TerminationSelection
+  -> IO (ServerResponse 'Normal CommandQueryResponse)
+handleTerminatePlan sc ts = do
+  handleTerminate sc ts
```

Comment:
Instead of splitting this up, you could choose to use void in your code to ignore the returned value.

Reply:
I thought about that before. And I think split it will make code cleaner? Since use void will generate a useless command query response?

Reply:
And I'm wondering why I can't assign you as reviewers... I wanted to request you to review this directly...
Your name just not show up in the reviewers list, even I full type your ID, it won't show up...

Reply:
> I thought about that before. And I think split it will make code cleaner? Since use void will generate a useless command query response?

I don't like it this way.

Reply:
> I thought about that before. And I think split it will make code cleaner? Since use void will generate a useless command query response?

Keep `handleTerminate`, and move the body of `handleTerminatePlan` back to executeQueryHandler`

Reply:
Done